src/int.rs

```diff
@@ -0,0 +1,266 @@
+//! This module contains an 256-bit signed integer implementation.
+
+use crate::errors::ParseI256Error;
+use std::fmt;
+use std::str::{self, FromStr};
+use web3::types::U256;
+
+/// Compute the two's complement of a U256.
+fn twos_complement(u: U256) -> U256 {
+    let (twos_complement, _) = (u ^ U256::max_value()).overflowing_add(U256::one());
+    twos_complement
+}
+
+/// Little-endian 256-bit signed integer.
+#[derive(Clone, Copy, Default, Eq, Hash, PartialEq)]
+pub struct I256(U256);
+
+/// Enum to represent the sign of a 256-bit signed integer.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum Sign {
+    /// Greater than or equal to zero.
+    Positive,
+    /// Less than zero.
+    Negative,
+}
+
+impl I256 {
+    /// Creates an I256 from a sign and an absolute value.
+    fn overflowing_from_sign_and_abs(sign: Sign, abs: U256) -> (Self, bool) {
+        let value = I256(match sign {
+            Sign::Positive => abs,
+            Sign::Negative => twos_complement(abs),
+        });
+        (value, value.sign() != sign)
+    }
+
+    /// Creates an I256 from an absolute value and a negative flag. Returns
+    /// `None` if it would overflow an `I256`.
+    fn checked_from_sign_and_abs(sign: Sign, abs: U256) -> Option<Self> {
+        let (result, overflow) = I256::overflowing_from_sign_and_abs(sign, abs);
+        if overflow {
+            None
+        } else {
+            Some(result)
+        }
+    }
+
+    /// Splits a I256 into its absolute value and negative flag.
+    fn into_sign_and_abs(self) -> (Sign, U256) {
+        let sign = self.sign();
+        let abs = match sign {
+            Sign::Positive => self.0,
+            Sign::Negative => twos_complement(self.0),
+        };
+        (sign, abs)
+    }
+
+    /// Returns an `i64` representing the sign of the number.
+    fn signum64(self) -> i64 {
+        let most_significant_word = (self.0).0[3] as i64;
+        most_significant_word.signum()
+    }
```

Comment:
If the last i64 is zero but a previous i64 is nonzero then the number is negative but the function returns 0 instead of -1. Maybe reuse `into_sign_and_abs` here.

Reply:
I think you are looking at an old commit.